### PR TITLE
gl: Enable core context and debug output.

### PR DESCRIPTION
**Pull request by vlj**

### CODE REVIEW

**Comment by Bigpet** [comment]
Just to note, a debug context does make some opengl drivers significantly slower, even without synchronous output. 

Also, some more filtering by severity will probably useful, since some drivers swamp the debug output with performance hints.

**Comment by kd-11** [comment]
Core profile should help with most gpu debuggers. Should we also move shader validation to an if (debug) block or not? It should in theory only be called once per shader (first instance it is used) anyway and saps away at performance.

File: rpcs3/Gui/GLGSFrame.cpp [hunk 1]
```diff
@@ -1,18 +1,41 @@
...
-	};
+		const int context_attrs[] =
+		{
+			WX_GL_RGBA,
+			WX_GL_DEPTH_SIZE, 16,
+			WX_GL_DOUBLEBUFFER,
+			WX_GL_MAJOR_VERSION, 3,
+			WX_GL_MINOR_VERSION, 3,
+			WX_GL_CORE_PROFILE,
+			WX_GL_DEBUG,
```
Comment (Nekotekina) [inline rpcs3/Gui/GLGSFrame.cpp]: Could just write conditional there.

Comment (Nekotekina) [inline rpcs3/Gui/GLGSFrame.cpp]:
```C++
rpcs3::config.rsx.d3d12.debug_output.value() ? WX_GL_DEBUG : 0
```

Comment (kd-11) [inline rpcs3/Gui/GLGSFrame.cpp]: @Nekotekina That parameter does not take a value from looking at the sources. wxwidgets is inconsistent like that.

Comment (Nekotekina) [inline rpcs3/Gui/GLGSFrame.cpp]: @kd-11 I totally don't understand what you mean. I also tried my suggestion and it compiles fine.

**Comment by vlj** [comment]
Use conditionnal operator and rebased.

File: rpcs3/Gui/GLGSFrame.cpp [hunk 2]
```diff
@@ -9,10 +11,14 @@ GLGSFrame::GLGSFrame() : GSFrame("OpenGL")
...
+		WX_GL_MAJOR_VERSION, 3,
+		WX_GL_MINOR_VERSION, 3,
+		WX_GL_CORE_PROFILE,
+#if wxCHECK_VERSION(3, 1, 0)
+		rpcs3::config.rsx.d3d12.debug_output.value() ? WX_GL_DEBUG : 0,
+#endif
 		0
 	};
-
-	m_canvas = new wxGLCanvas(this, wxID_ANY, context_attrs);
```
Comment (Nekotekina) [inline rpcs3/Gui/GLGSFrame.cpp]: String got removed for some reason.

**Comment by vlj** [comment]
I don't understand wxCheck_version returns true when wx_debug is not defined.

**Pull request merged**